### PR TITLE
EC-904 Filter by commenter

### PR DESCRIPTION
**Pull request by suegarner**
https://nicedigital.atlassian.net/browse/EC-904

### CODE REVIEW

File: Comments/Models/Context.cs [hunk 1]
```diff
@@ -965,5 +965,21 @@ public bool AreAnswersForThisOrganisation(List<int> answerIds, int organisationI
...
-	}
+
+        public List<string> GetEmailAddressForCommentsAndAnswers(CommentsAndQuestions commentsAndQuestions)
+        {
+            var commentIds = commentsAndQuestions.Comments.Select(c => c.CommentId).ToList();
+            var answerIds = commentsAndQuestions.Questions.SelectMany(q => q.Answers).Select(a => a.AnswerId).ToList();
+
+
+            var emailAddresses = OrganisationUser
+                .Where(o => Comment.Where(c => commentIds.Contains(c.CommentId)).Select(c => c.OrganisationUserId).Contains(o.OrganisationUserId)
```
Comment (John-Williams) [inline Comments/Models/Context.cs:976]: regarding the use of Comment and Answer subqueries here...

Is is all handled by the database, or is it brought back to the application server?

Also, does it requery the comment and answer tables repeatedly for each organisation user?

If so, i think it needs changing.

File: Comments/Services/CommentService.cs [hunk 2]
```diff
@@ -230,10 +230,28 @@ public CommentsAndQuestions FilterCommentsAndQuestions(CommentsAndQuestions comm
 			    commentsAndQuestions.Questions.ForEach(q => q.Show = (!q.Show || !q.DocumentId.HasValue) ? false : documentIds.Contains(q.DocumentId.Value));
 			    commentsAndQuestions.Comments.ForEach(c => c.Show = (!c.Show || !c.DocumentId.HasValue) ? false : documentIds.Contains(c.DocumentId.Value));
 			}
+
+			var commenters = commentersToFilter?.ToList() ?? new List<string>(); ;
+			if (commenters.Any())
+			{
+				commentsAndQuestions.Comments.ForEach(c => c.Show = (!c.Show || c.CommenterEmail.Length == 0) ? false : commenters.Contains(c.CommenterEmail));
```
Comment (John-Williams) [inline Comments/Services/CommentService.cs:237]: is the email ever lowercased? if so the contains needs to support  that. 

however if we store the email address and display it again in the same case it was input this is fine.

**Review by John-Williams — CHANGES_REQUESTED** [review]
Looks good, thanks!

I've just asked a couple of questions inline. cheers.

File: Comments/Services/CommentService.cs (hunk 2 above)
Comment (suegarner) [reply Comments/Services/CommentService.cs:237]: Added StringComparer.OrdinalIgnoreCase to the contains

File: Comments/Models/Context.cs (hunk 1 above)
Comment (suegarner) [reply Comments/Models/Context.cs:976]: Need to chat to you about this

**Review by John-Williams — APPROVED** [review]

**Review by emollett — DISMISSED** [review]

**Review by emollett — APPROVED** [review]

**Pull request merged**